### PR TITLE
Display alert messages on dashboard from local settings

### PR DESCRIPTION
**Pull request by jedcunningham**
Allow alert messages to be displayed on the UI dashboard from
`airflow_local_settings`. This feature can be used to announce things to
all users, warn about setup limitations/issues, and probably other
things I haven't considered.

The message, category, and role(s) the flash message should be shown to
can be configured.

Here is an example:

In `airflow_local_settings`:
```
from airflow.www.utils import UIAlert
DASHBOARD_UIALERTS = [
    UIAlert("hello", category="error"),
    UIAlert("world", roles=["Viewer", "Admin"]),
    UIAlert('Visit <a href="https://airflow.apache.org">airflow.apache.org</a>', html=True),
    UIAlert("Only for users", roles=["Users"]),
]
```

View as an Admin user:
Screen Shot 2021-09-15 at 5 28 49 PM

View as a User user:
Screen Shot 2021-09-15 at 5 35 42 PM

related: #18098

### CODE REVIEW

File: airflow/settings.py [hunk 1]
```diff
@@ -519,3 +520,13 @@ def initialize():
...
 MASK_SECRETS_IN_LOGS = False
+
+# Display flash messages on the dashboard
+# Useful for warning about setup issues or announcing changes to end users
+# You can choose what role(s) the message will be shown to as well, or pass None to show to all users
+# (message, category, [role(s)]), for example:
+# ("Airflow update happening next week", "warning", ["User"])
+# You can also pass the message as flask.Markup to include HTML in the message:
+# (Markup('Visit <a href="http://airflow.apache.org">airflow.apache.org</a>'), "info", None)
+FlashMessage = Tuple[Union[str, Markup], str, Optional[str]]
```
Comment (jedcunningham) [inline airflow/settings.py]: I considered creating a class so we can simplify the experience here a bit, e.g. something like:

```
DASHBOARD_FLASH_MESSAGES = [
    FlashMessage("Hello"),
    FlashMessage("Word", roles=["Admin"]),
    FlashMessage("example", category="Warning"),
]
```

Thoughts?

Comment (uranusjr) [reply airflow/settings.py]: A custom class feels like a good idea to me. We can also prevent explicitly depending on `Markup` for simple cases e.g. `FlashMessage("<a href="example.com">Example</a>", html=True)`. Although `Markup` would still be needed if the user needs more elaborated stuff like `.format()`.

Speaking of `format()`, we should add to the documentation (which I assume will come later) the implied danger using raw HTML and how to do thing property (I think MarkupSafe has some writeup on this). Maybe we should also ask Andrew how these things are handled in Django-land.

**Review by kaxil — COMMENTED** [review]
looks good -- we should add docs on it too

File: airflow/settings.py (hunk 1 above)
Comment (jedcunningham) [reply airflow/settings.py]: Thanks @uranusjr, good feedback and I think I've covered it all at this point.

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Unrelated test failures:

```
=========================== short test summary info ============================
  FAILED tests/kubernetes/test_pod_generator.py::TestPodGenerator::test_construct_pod
  FAILED tests/kubernetes/test_pod_generator.py::TestPodGenerator::test_construct_pod_empty_executor_config
  FAILED tests/kubernetes/test_pod_generator.py::TestPodGenerator::test_deserialize_model_file
  FAILED tests/kubernetes/test_pod_generator.py::TestPodGenerator::test_gen_pod_extract_xcom
  FAILED tests/kubernetes/test_pod_generator.py::TestPodGenerator::test_reconcile_pods
  FAILED tests/kubernetes/test_pod_generator.py::TestPodGenerator::test_reconcile_pods_empty_mutator_pod
  ======= 6 failed, 506 passed, 15 skipped, 1 warning in 495.43s (0:08:15) =======
  
```

**Comment by uranusjr** [comment]
```
parameters-ref.rst:2472: (pathType)  The pathType for the web Ingress (required for Kubernetes 1.19 and above).

File path: helm-chart/parameters-ref.rst
Incorrect Spelling: 'pathType'
Line with Error: 'The pathType for the web Ingress (required for Kubernetes 1.19 and above).'
Line Number: 2472
```

**Comment by kaxil** [comment]
> ```
> parameters-ref.rst:2472: (pathType)  The pathType for the web Ingress (required for Kubernetes 1.19 and above).
> 
> File path: helm-chart/parameters-ref.rst
> Incorrect Spelling: 'pathType'
> Line with Error: 'The pathType for the web Ingress (required for Kubernetes 1.19 and above).'
> Line Number: 2472
> ```

That should be fixed by https://github.com/apache/airflow/pull/18308

**Comment by jedcunningham** [comment]
The unrelated `test_pod_generator` failures should be fixed with #18384.

File: docs/apache-airflow/howto/customize-ui.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+-------------------------------------------------
+
+Airflow now allows you to customize the DAG home page header and page title. This will help
+distinguish between various installations of Airflow or simply amend the page text.
+
+Note: the custom title will be applied to both the page header and the page title.
+
+To make this change, simply:
+
+1.  Add the configuration option of ``instance_name`` under ``webserver`` inside ``airflow.cfg``:
```
Comment (mik-laj) [inline docs/apache-airflow/howto/customize-ui.rst:91]: Section should have `[]` for consistency.

Comment (jedcunningham) [reply docs/apache-airflow/howto/customize-ui.rst:91]: Thanks @mik-laj, opened #18658 to fix it.

**Pull request merged**